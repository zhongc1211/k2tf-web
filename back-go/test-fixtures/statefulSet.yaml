apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: web
  labels:
    app: nginx
spec:
  serviceName: "nginx"
  selector:
    matchLabels:
      app: nginx
  replicas: 14
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
        - name: nginx
          image: k8s.gcr.io/nginx-slim:0.8
          ports:
            - containerPort: 80
              name: web
          volumeMounts:
            - name: www
              mountPath: /usr/share/nginx/html
  updateStrategy:
    type: RollingUpdate
  volumeClaimTemplates:
    - metadata:
        name: www
      spec:
        accessModes: ["ReadWriteOnce"]
        resources:
          requests:
            storage: 1Gi
        storageClassName: thin-disk
