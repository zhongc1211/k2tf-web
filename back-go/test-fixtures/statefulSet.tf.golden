resource "kubernetes_stateful_set" "web" {
  metadata {
    name   = "web"
    labels = { app = "nginx" }
  }
  spec {
    replicas = 14
    selector {
      match_labels = { app = "nginx" }
    }
    template {
      metadata {
        labels = { app = "nginx" }
      }
      spec {
        container {
          name  = "nginx"
          image = "k8s.gcr.io/nginx-slim:0.8"
          port {
            name           = "web"
            container_port = 80
          }
          volume_mount {
            name       = "www"
            mount_path = "/usr/share/nginx/html"
          }
        }
      }
    }
    volume_claim_template {
      metadata {
        name = "www"
      }
      spec {
        access_modes = ["ReadWriteOnce"]
        resources {
          requests = { storage = "1Gi" }
        }
        storage_class_name = "thin-disk"
      }
    }
    service_name = "nginx"
    update_strategy {
      type = "RollingUpdate"
    }
  }
}
